### PR TITLE
feat(frontend-canister): pretty print asset properties during deployment

### PR DESCRIPTION
**Pull request by smallstepman**
# Description

- pretty print AssetProperties when deploying assets

# How Has This Been Tested?

e2e 

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I have edited the CHANGELOG accordingly.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: src/distributed/assetstorage.did [hunk 1]
```diff
@@ -6,6 +6,10 @@ type Time = int;
 type CreateAssetArguments = record {
   key: Key;
   content_type: text;
+  properties: AssetProperties;
+};
+
+type AssetProperties = record {
```
Comment (ericswanson-dfinity) [inline src/distributed/assetstorage.did:12]: This is a breaking change at the interface level.  Is this intentional?

File: src/canisters/frontend/ic-certified-assets/src/state_machine.rs [hunk 2]
```diff
@@ -36,12 +36,17 @@ pub struct AssetEncoding {
...
+    pub headers: Option<HashMap<String, String>>,
+}
+
 #[derive(Default, Clone, Debug, CandidType, Deserialize)]
 pub struct Asset {
     pub content_type: String,
     pub encodings: HashMap<String, AssetEncoding>,
-    pub max_age: Option<u64>,
-    pub headers: Option<HashMap<String, String>>,
+    pub properties: AssetProperties,
```
Comment (ericswanson-dfinity) [inline src/canisters/frontend/ic-certified-assets/src/state_machine.rs:49]: Asset is a member of StableState.  What's the plan for upgradability?

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
At a minimum we should talk about the impact of breaking interface changes before moving forward, if that's intentional

File: src/distributed/assetstorage.did (hunk 1 above)
Comment (smallstepman) [reply src/distributed/assetstorage.did:12]: thank you, and no -- I completely forgot to think about this quite important detail
let's discuss this during stand up

File: src/canisters/frontend/ic-certified-assets/src/state_machine.rs (hunk 2 above)
Comment (smallstepman) [reply src/canisters/frontend/ic-certified-assets/src/state_machine.rs]: I decided to make this PR only about pretty printing 

(also smuggling small update to include `aliasing` in `get_/set_asset_properties`)

File: src/distributed/assetstorage.did (hunk 1 above)
Comment (ericswanson-dfinity) [reply src/distributed/assetstorage.did:12]: This is still a breaking change at the interface level.  So at the moment, this PR is more than just "pretty print asset properties during deployment."  Am I forgetting a resolution with regard to this?

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
Please move all of the pieces that are not related to pretty-printing existing data into a separate PR.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 ## DFX
 
+### feat(frontend-canister): pretty print asset properties when deploing assets to the canister
```
Comment (ericswanson-dfinity) [inline CHANGELOG.md:7]:
```suggestion
### feat(frontend-canister): pretty print asset properties when deploying assets to the canister
```

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**